src/overlays/actors/ovl_En_Toryo/z_en_toryo.c

```diff
@@ -29,27 +30,343 @@ const ActorInit En_Toryo_InitVars = {
+            this->actor.textId = func_80B206A0(this, globalCtx);
+            func_8002F298(&this->actor, globalCtx, 100.0f, 10);
+        }
+    }
+}
+
+void func_80B20914(EnToryo* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    func_80B20768(this, globalCtx);
+    if (this->unk_1E4 != 0) {
+        this->stateFlags |= 0x10;
+    } else {
+        this->stateFlags &= ~0x10;
+    }
+}
+
+void EnToryo_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnToryo* this = THIS;
+    ColliderCylinder* collider = &this->collider;
+    Player* player = PLAYER;
+    f32 rot;
+
+    Collider_CylinderUpdate(thisx, collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, (Collider*)collider);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B203D8.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B205CC.s")
+    if ((this->stateFlags & 0x8)) {
+        this->unk_1EC.unk_18.x = player->actor.posRot2.pos.x;
+        this->unk_1EC.unk_18.y = player->actor.posRot2.pos.y;
+        this->unk_1EC.unk_18.z = player->actor.posRot2.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B20634.s")
+        if ((this->stateFlags & 0x10)) {
+            func_80034A14(thisx, &this->unk_1EC, 0, 4);
+            return;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B206A0.s")
+        rot = thisx->yawTowardsLink - thisx->shape.rot.y;
+        if ((rot < 14563.0f) && (rot > -14563.0f)) {
+            func_80034A14(thisx, &this->unk_1EC, 0, 2);
+        } else {
+            func_80034A14(thisx, &this->unk_1EC, 0, 1);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B20768.s")
+void EnToryo_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnToryo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/func_80B20914.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                          EnToryo_OverrideLimbDraw, EnToryo_PostLimbDraw, this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/EnToryo_Update.s")
+s32 EnToryo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             void* thisx) {
+    EnToryo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Toryo/EnToryo_Draw.s")
+    if ((this->stateFlags & 0x8)) {
+        switch (limbIndex) {
+            case 8:
+                rot->x += this->unk_1EC.unk_0E.y;
+                rot->y -= this->unk_1EC.unk_0E.x;
+                break;
+            case 15:
+                rot->x += this->unk_1EC.unk_08.y;
+                rot->z += this->unk_1EC.unk_08.x;
```

Comment:
I suggest having a break in last cases like this one, for consistency and to avoid mistakes when adding or moving cases during modding. Probably relevant in other switches in this file
```suggestion
                rot->z += this->unk_1EC.unk_08.x;
                break;
```